lib/functions/OverloadTree.ts

```diff
@@ -131,7 +130,7 @@ export class OverloadTree {
     for (const term of args) {
       const literalExpression = isLiteralTermExpression(term);
       const key = literalExpression ? literalExpression.dataType : term.termType;
-      cache.cache = {};
```

Comment:
I altered this line because it basically reseted some parts of the cache sometimes. (I re-checked whether the cache size still had manageable size and all seemed fine :)